src/controllers/portfolio/portfolio.ts

```diff
@@ -645,6 +665,29 @@ export class PortfolioController extends EventEmitter {
       : Promise.resolve()
 
     const networks = network ? [network] : this.#networks.networks
+
+    let hintsFromExternalAPIByNetworks: ExternalHintsAPIResponse[] = []
+    let hintsFromExternalAPIError: null | ExtendedErrorWithLevel = null
+
+    try {
+      const networkIds = networks.filter(({ hasRelayer }) => hasRelayer).map((x) => x.id)
+      const hintsFromExternalAPIByNetworksData = await this.#fetch(
```

Comment:
I propose we use the batcher here. We already use it in the portfolio lib, maybe reuse it in a file as a lib as velcroDiscoveryBatcher